docs/debug_virt_stack/launch-qemu-strace.md

```diff
@@ -0,0 +1,215 @@
+
+Once all the objects created, we can finally run the guest to debug.
+
+VMI:
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  annotations:
+    hooks.kubevirt.io/hookSidecars: '[{"args": ["--version", "v1alpha2"],
+    "image":"registry:5000/kubevirt/sidecar-shim:devel",
+    "pvc": {"name": "debug-tools","volumePath": "/debug", "sharedComputePath": "/var/run/debug"},
+    "configMap": {"name": "my-config-map","key": "my_script.sh", "hookPath": "/usr/bin/onDefineDomain"}}]'
+  labels:
+    special: vmi-debug-tools
+  name: vmi-debug-tools
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+      - disk:
+          bus: virtio
+        name: cloudinitdisk
+      rng: {}
+    resources:
+      requests:
+        memory: 1024M
+  terminationGracePeriodSeconds: 0
+  volumes:
+  - containerDisk:
+      image: registry:5000/kubevirt/fedora-with-test-tooling-container-disk:devel
+    name: containerdisk
+  - cloudInitNoCloud:
+      userData: |-
+        #cloud-config
+        password: fedora
+        chpasswd: { expire: False }
+    name: cloudinitdisk
+```
+
+The VMI example is a simply VM instance declaration and the interesting parts are the annotations for the hook:
+* `image` refers to the sidecar-shim already built and shipped by KubeVirt release
+* `pvc` refers to the PVC populated with the debug setup. The `name` refers to the claim name, the `volumePath` is the path inside the sidecar container where the volume is mounted while the `sharedComputePath` is the path of the same volume inside the compute container.
+* `configMap` refers to the confimap containing the script to modify the XML for the wrapping script
+
+Once the VM is declared, the hook will modify the emulator section and Libvirt will call the wrapping script instead of QEMU directly.
+
+## How to fetch the output
+
+The wrapping script configure `strace` to store the output in the PVC. In this way, it is possible to retrieve the output file in a later time, for example using an additional pod like:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: fetch-logs
+spec:
+  securityContext:
+    runAsUser: 107
+    fsGroup: 107
+  volumes:
+    - name: populate
+      persistentVolumeClaim:
+        claimName: debug-tools
+  containers:
+    - name: populate
+      image: busybox:latest
+      command: ["tail", "-f", "/dev/null"]
+      volumeMounts:
+        - mountPath: "/vol"
+          name: populate
+```
+
+and it possible to copy the file locally with:
+```bash
+$ kubectl cp fetch-logs:/vol/logs/strace.out strace.out
+```
+
+You can see a full demo of this setup:
+[![asciicast](https://asciinema.org/a/1Cm87DcjtUWprRrWPBzKfxVmp.svg)](https://asciinema.org/a/1Cm87DcjtUWprRrWPBzKfxVmp)
```

Comment:
:star_struck: 
To many, that might be the preferred way to consume this howto. Might be worth mentioning the existence of the demo at the beginning of the document :)